qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1269,8 +1278,17 @@ def nodes_on_wire(self, wire, only_ops=False):
                 yield current_node
 
             # find the adjacent node that takes the wire being looked at as input
-            for node, edges in self._multi_graph.adj[current_node].items():
-                if any(wire == edge['wire'] for edge in edges.values()):
+            # TODO(mtreinish): Add function in retworkx that does this nested api
```

Comment:
Is this fixed with `.adj_direction(direction=False)`?

Reply:
Unfortunately it doesn't because of multi graphs, `adj_direction` uses petgraph's `find_edge` method currently (https://docs.rs/petgraph/0.5.0/petgraph/stable_graph/struct.StableGraph.html#method.find_edge ) Which only returns a single edge. This needs to use all the edges which is why it loops over the adjacent nodes and calls `get_all_edge_data` to get the value of all the edges, even for multigraphs.

That being said @gadial has a PR open starting to work on this here: https://github.com/mtreinish/retworkx/pull/25 I haven't had a chance to review it in depth yet though.